centreon/src/Core/Dashboard/Application/UseCase/FindDashboard/FindDashboardFactory.php

```diff
@@ -69,6 +75,26 @@ public static function createResponse(
             $response->updatedBy->name = $contactNames[$contactId]['name'] ?? '';
         }
 
+        if ($ownRole === DashboardSharingRole::Editor && array_key_exists($dashboard->getId(), $contactShares)) {
+            $response->shares['contacts'] = array_map(static fn (DashboardContactShare $contactShare): array => [
+                'id' => $contactShare->getContactId(),
+                'name' => $contactShare->getContactName(),
+                'email' => $contactShare->getContactEmail(),
+                'role' => $contactShare->getRole(),
+            ],
+            $contactShares[$dashboard->getId()]);
+        }
+
+        if ($ownRole === DashboardSharingRole::Editor && array_key_exists($dashboard->getId(), $contactGroupShares)) {
+            $response->shares['contact_groups'] = array_map(
+                static fn (DashboardContactGroupShare $contactGroupShare): array => [
+                    'id' => $contactGroupShare->getContactGroupId(),
+                    'name' => $contactGroupShare->getContactGroupName(),
+                    'role' => $contactGroupShare->getRole(),
+                ],
+                $contactGroupShares[$dashboard->getId()]);
+        }
+
```

Comment:
You could add a few comments to explicit the functional behaviour wanted for this